source/AI.cpp

```diff
@@ -2854,6 +2854,31 @@ Point AI::TargetAim(const Ship &ship, const Body &target)
 	return result ? result : target.Position() - ship.Position();
 }
 
+bool AI::TargetInRange(const Ship &ship, const Hardpoint &hardpoint)
```

Comment:
If I recall correctly, a part of the functions that determines if a wepaon should fire uses a similiar method, maybe this function can also be used there?

Reply:
Had the same thought and took a look around and couldn't find any place to use this directly.

There is similar logic in a number of places in AI.ccp, but they are for checking if all `ANY` possible hard points are in range, not `EACH` hard point. Using this there would waste CPU cycles.